src/store/modules/emails.js

```diff
@@ -11,18 +10,16 @@ const getters = {
     //emails will be set within state from emails service
     return state.mailObject;
   },
-  getSelectedSenderObject(state, getters) {
+  getMailObjectById: (state, getters) => (id) => {
     //finds the emails attached to the selected sender's ID
-    return getters.getMailObject.find(
-      (sender) => sender.senderId === state.selectedSenderId
-    );
+    return getters.getMailObject.find((sender) => sender.id === id);
   },
   //returns all the emails, with most recent ones first
-  getEmailsOrderByDateAsc(state, getters) {
+  getMailObjectEmailsOrderByDateAsc: (state, getters) => (id) => {
     //get list of emails
-    const mail = getters.getSelectedSenderObject.emails;
+    var mailObject = { ...getters.getMailObjectById(id) };
     //sorts a copy of the list of mail (getters should not mutate state)
-    const sortedMail = [...mail].sort((a, b) => {
+    mailObject.emails = [...mailObject.emails].sort((a, b) => {
```

Comment:
Maybe we should have some error handling here just like the chatMessages.js?

What if mailObject is undefined?

Reply:
I must have added it in while you were commenting on this PR. However! I did add a check to make sure some emails existed like in chatMessages